src/test.sml

```diff
@@ -1,7 +1,23 @@
 structure Test : TEST =
 struct
+
+  val testsRan = ref 0
+  val testsPassed = ref 0
+
   exception Failed
 
   fun test f = f () handle Failed => false
-  fun assert b = if b then true else raise Failed
+
+  fun assert(expected,actual) =
+    (testsRan := !testsRan + 1;
+     if expected = actual
+     then (testsPassed := !testsPassed + 1; true)
+     else raise Failed)
```

Comment:
Also considering we are using exceptions here, the return type of `bool` is pretty meaningless. We could get away with a unit return type.